crates/cli/src/lib.rs

```
@@ -221,45 +233,63 @@ impl Install {
-
-                new_lines.push(line);
-            }
-
-            // Comment out extension if user specifies disable flag
-            if self.disable {
-                ext_line.insert(0, ';');
-            }
-
-            new_lines.push(ext_line);
-            file.rewind()?;
-            file.set_len(0)?;
-            file.write(new_lines.join("\n").as_bytes())
+            update_ini_file(&php_ini, ext_name, self.disable)
                 .with_context(|| "Failed to update `php.ini`")?;
         }
 
         Ok(())
     }
 }
 
+// Update extension information in the ini file, if fails, try with sudo again.
+//
+// Write to a temp file then move it to given path. Use `sudo` on unix to move
+// file if needed.
+fn update_ini_file(php_ini: &PathBuf, ext_name: &str, disable: bool) -> anyhow::Result<()> {
+    let current_ini_content = std::fs::read_to_string(php_ini)?;
+    let mut ext_line = format!("extension={ext_name}");
+
+    let mut new_lines = current_ini_content.lines().collect::<Vec<_>>();
+    for line in &new_lines {
+        if line.contains(&ext_line) {
+            bail!("Extension already enabled.");
+        }
+    }
+
+    // Comment out extension if user specifies disable flag
+    if disable {
+        ext_line.insert(0, ';');
+    }
+
+    new_lines.push(&ext_line);
+    write_to_file(new_lines.join("\n"), php_ini)?;
+    Ok(())
+}
+
+// Copy extension, if fails, try with sudo cp.
+//
+// Checking if we have write permission for ext_dir may fail due to ACL, group
+// list and and other nuances. See
+// https://doc.rust-lang.org/std/fs/struct.Permissions.html#method.readonly
+fn copy_extension(ext_path: &Utf8PathBuf, ext_dir: &PathBuf) -> anyhow::Result<()> {
+    if let Err(_e) = std::fs::copy(ext_path, ext_dir) {
+        #[cfg(unix)]
+        {
+            let _ = std::process::Command::new("sudo")
+                .arg("cp")
+                .arg(ext_path)
+                .arg(ext_dir)
+                .status()?;
+        }
```

This silently drops the error on non unix systems